keyboards/helix/rev3_5rows/keymaps/five_rows/oled_display.c

```diff
@@ -154,12 +223,10 @@ void render_status(void) {
     }
 
     // Host Keyboard LED Status
-    char led[40];
-    snprintf(led, sizeof(led), "\n%s  %s  %s",
-             (host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? "NUMLOCK" : "       ",
-             (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? "CAPS" : "    ",
-             (host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? "SCLK" : "    ");
-    oled_write(led, false);
+    oled_write_P(PSTR("\n"), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_NUM_LOCK)) ? PSTR("NUMLOCK  ") : PSTR("         "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) ? PSTR("CAPS  ") : PSTR("      "), false);
+    oled_write_P((host_keyboard_leds() & (1<<USB_LED_SCROLL_LOCK)) ? PSTR("SCLK  ") : PSTR("      "), false);
```

Comment:
```suggestion
    led_t led_state = host_keyboard_led_state();
    oled_write_P(PSTR("\n"), false);
    oled_write_P(led_state.num_lock ? PSTR("NUMLOCK  ") : PSTR("         "), false);
    oled_write_P(led_state.caps_lock ? PSTR("CAPS  ") : PSTR("      "), false);
    oled_write_P(led_state.scroll_lock ? PSTR("SCLK  ") : PSTR("      "), false);
```

Reply:
I think it's not a bad idea to get the state of each LED, such as `led_state.num_lock`.

However, I found a potential problem in `quantum/led.h`.
The `led_t` declaration is not portable code as far as I know.
Therefore, the following code in `tmk_core/common/host.c` is also not portable.
```c
led_t host_keyboard_led_state(void) { return (led_t)host_keyboard_leds(); }
```

Try compiling the code below.
```
$ cat bit_fields_test.c
#include "led.h"

led_t status1 = { .num_lock = 1 };
led_t status2 = { .raw = (1<<USB_LED_NUM_LOCK) };
```
In the following compilation, status1 and status2 have the same value.
```
$ arm-none-eabi-gcc -mlittle-endian -S bit_fields_test.c
$ grep -A1 ^status  bit_fields_test.s
status1:
	.byte	1
--
status2:
	.byte	1
```

In the following compilations status1 and status2 will not have the same value.
```
$ arm-none-eabi-gcc -mbig-endian -S bit_fields_test.c
$ grep -A1 ^status  bit_fields_test.s                
status1:
	.byte	128
--
status2:
	.byte	1
```

Reply:
AVR is little-endian, and it seems all the currently supported ARM MCUs default to little-endian, so I don't see this as a pressing issue just yet.

Reply:
Done.